File: LATPRAK2DPBO/Python/main.py
# Saya Marvel Ravindra Dioputra [2200481] LatPrak2 Python dalam Mata Kuliah DPBO 
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

# class parent dari class clothing
class Product:
    def __init__(self, idProduct, name, brand, price):
        self._idProduct = idProduct # inisialisasi atribut id produk
        self._name = name           # inisialisasi atribut nama produk
        self._brand = brand         # inisialisasi atribut merk produk
        self._price = price         # inisialisasi atribut harga produk

    # getter dan setter untuk atribut diatas
    def get_idProduct(self):
        return self._idProduct

    def set_idProduct(self, idProduct):
        self._idProduct = idProduct

    def get_name(self):
        return self._name

    def set_name(self, name):
        self._name = name

    def get_brand(self):
        return self._brand

    def set_brand(self, brand):
        self._brand = brand

    def get_price(self):
        return self._price

    def set_price(self, price):
        self._price = price

# class parent dari class shirt
class Clothing(Product):
    def __init__(self, idProduct, name, brand, price, size, material, gender):
        super().__init__(idProduct, name, brand, price) # hubungan inheritance dari parent
        self._size = size           # inisialisasi atribut ukuran
        self._material = material   # inisialisasi atribut material/bahan
        self._gender = gender       # inisialisasi atribut gender

    # getter dan setter untuk atribut diatas
    def get_size(self):
        return self._size

    def set_size(self, size):
        self._size = size

    def get_material(self):
        return self._material

    def set_material(self, material):
        self._material = material

    def get_gender(self):
        return self._gender

    def set_gender(self, gender):
        self._gender = gender

# class cucu dari class product
class Shirt(Clothing):
    def __init__(self, idProduct, name, brand, price, size, material, gender, color, sleeveType):
        super().__init__(idProduct, name, brand, price, size, material, gender) # hubungan inheritance dari parent
        self._color = color             # inisialisasi atribut warna
        self._sleeveType = sleeveType   # inisialisasi atribut jenis/panjang lengan

    # getter dan setter untuk atribut diatas
    def get_color(self):
        return self._color

    def set_color(self, color):
        self._color = color

    def get_sleeveType(self):
        return self._sleeveType

    def set_sleeveType(self, sleeveType):
        self._sleeveType = sleeveType

# main
def main():
    shirts = [] # list untuk menampung data inputan user
    number_of_shirts = int(input("Masukan jumlah baju: "))  # masukan user untuk jumlah data yang diinginkan

    # perulangan untuk memasukan data untuk setiap objek "shirt"
    for i in range(number_of_shirts):
        print(f"Masukan Data {i + 1}:")
        id_ = input("ID: ")
        name = input("Name: ")
        brand = input("Brand: ")
        price = input("Price: ")
        size = input("Size: ")
        material = input("Material: ")
        gender = input("Gender: ")
        color = input("Color: ")
        sleeveType = input("Sleeve Type: ")

        shirt = Shirt(id_, name, brand, price, size, material, gender, color, sleeveType)
        shirts.append(shirt)

    # output data yang telah diinput dalam tabel
    print("+----+--------------+--------------+--------------+--------------+--------------+--------------+---------+-------------+")
    print("| No |      ID      |     Name     |     Brand    |    Color     |    Price     |     Size     | Gender  | Sleeve Type |")
    print("+----+--------------+--------------+--------------+--------------+--------------+--------------+---------+-------------+")

    for i, shirt in enumerate(shirts, start=1):
        print(f"| {i:>2} | {shirt.get_idProduct():>12} | {shirt.get_name():>12} | {shirt.get_brand():>12} | {shirt.get_color():>12} | {shirt.get_price():>12} | {shirt.get_size():>12} | {shirt.get_gender():>7} | {shirt.get_sleeveType():>11} |")

    print("+----+--------------+--------------+--------------+--------------+--------------+--------------+---------+-------------+")

if __name__ == "__main__":
    main()

File: LATPRAK2DPBO/C++/main.cpp
// Saya Marvel Ravindra Dioputra [2200481] LatPrak2 Python dalam Mata Kuliah DPBO 
// untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

#include <iostream> // library input output
#include <iomanip>  // library untuk setw (panjang pada tabel)
#include <vector>   // library untuk vector/list penampung
#include <string>   // library untuk tipe data strng

using namespace std;

// class parent dari class Product
class Product {
protected:
    string idProduct;   // inisialisasi atribut id produk
    string name;        // inisialisasi atribut nama produk
    string brand;       // inisialisasi merk produk
    string price;       // inisialisasi harga produk

public:
    Product(string id, string n, string b, string p) : idProduct(id), name(n), brand(b), price(p) {}

    // getter dan setter
    string getIdProduct() { return idProduct; }
    void setIdProduct(string id) { idProduct = id; }

    string getName() { return name; }
    void setName(string n) { name = n; }

    string getBrand() { return brand; }
    void setBrand(string b) { brand = b; }

    string getPrice() { return price; }
    void setPrice(string p) { price = p; }
};

// class parent dari class Clothing
class Clothing : public Product {
protected:
    string size;        // inisialisasi atribut ukuran
    string material;    // inisialisasi atribut bahan
    string gender;      // inisialisasi atribut gender

public:
    Clothing(string id, string n, string b, string p, string s, string m, string g) : Product(id, n, b, p), size(s), material(m), gender(g) {}

    //getter dan setter
    string getSize() { return size; }
    void setSize(string s) { size = s; }

    string getMaterial() { return material; }
    void setMaterial(string m) { material = m; }

    string getGender() { return gender; }
    void setGender(string g) { gender = g; }
};

// class cucu dari class Product
class Shirt : public Clothing {
protected:
    string color;       // inisialisasi atribut warna
    string sleeveType;  // inisialisasi atribut panjang/jenis lengan

public:
    Shirt(string id, string n, string b, string p, string s, string m, string g, string c, string st) : Clothing(id, n, b, p, s, m, g), color(c), sleeveType(st) {}

    //getter dan setter
    string getColor() { return color; }
    void setColor(string c) { color = c; }

    string getSleeveType() { return sleeveType; }
    void setSleeveType(string st) { sleeveType = st; }
};

// main
int main() {
    vector<Shirt> shirts;
    int jumlah;
    cout << "Masukan jumlah baju: ";
    cin >> jumlah;

    //input data produk
    for (int i = 0; i < jumlah; i++) {
        cout << "Masukan Data " << i + 1 << ":" << endl;
        string id, name, brand, size, material, gender, color, sleeveType, price;

        cout << "ID: "; cin >> id;
        cout << "Name: "; cin >> name;
        cout << "Brand: "; cin >> brand;
        cout << "Price: "; cin >> price;
        cout << "Size: "; cin >> size;
        cout << "Material: "; cin >> material;
        cout << "Gender: "; cin >> gender;
        cout << "Color: "; cin >> color;
        cout << "Sleeve Type: "; cin >> sleeveType;

        Shirt shirt(id, name, brand, price, size, material, gender, color, sleeveType);
        shirts.push_back(shirt);
    }

    //output data pada list dalam tabel
    cout << "+----+--------------+--------------+--------------+--------------+--------------+--------------+---------+-------------+" << endl;
    cout << "| No |      ID      |     Name     |     Brand    |    Color     |    Price     |     Size     | Gender  | Sleeve Type |" << endl;
    cout << "+----+--------------+--------------+--------------+--------------+--------------+--------------+---------+-------------+" << endl;

    for (int i = 0; i < shirts.size(); i++) {
        Shirt shirt = shirts[i];
        cout << "| " << setw(2) << i + 1 << " | " << setw(12) << shirt.getIdProduct() << " | " << setw(12) << shirt.getName() << " | " << setw(12) << shirt.getBrand() << " | " << setw(12) << shirt.getColor() << " | " << setw(12) << shirt.getPrice() << " | " << setw(12) << shirt.getSize() << " | " << setw(7) << shirt.getGender() << " | " << setw(11) << shirt.getSleeveType() << " |" << endl;
    }

    cout << "+----+--------------+--------------+--------------+--------------+--------------+--------------+---------+-------------+" << endl;

    return 0;
}
